src/client.cr

```diff
@@ -27,34 +27,39 @@ module AWS
       "Connection" => "keep-alive",
       "User-Agent" => "Crystal AWS #{VERSION}",
     }
+
+    def encode(path : String) : String
+      URI.encode_path path
+    end
```

Comment:
I broke out this logic into a dedicated method because it sure seems like there are some atypical encoding rules in this world, and I'm sure it'll need to be special cased.

Also, if someone finds this is insufficient for what they need, it can be easily monkey-patched over.